[joy-ui][docs] Add Next.js App Router instructions on how to prevent flickering (#42474)
mui/material-ui: docs/data/joy/customization/dark-mode/dark-mode.md

  }
}
```

### Next.js App Router

> To utilize the Joy UI API with an App Router in a Next.js project, you need to create a separate [Client Component](https://nextjs.org/docs/app/building-your-application/rendering/client-components) in order to utilize [`getInitColorSchemeScript`](https://mui.com/joy-ui/main-features/dark-mode-optimization/#the-solution-css-variables):

Reviewer: ```suggestion
To use the Joy UI API with a Next.js project with the App Router, create a separate [Client Component](https://nextjs.org/docs/app/building-your-application/rendering/client-components) to utilize the [`getInitColorSchemeScript`](https://mui.com/joy-ui/main-features/dark-mode-optimization/#the-solution-css-variables) function:
```

I'd go with something like this!
